lib/puppet/provider/registry_value/registry.rb

```diff
@@ -95,8 +86,8 @@ def regvalue
     unless @regvalue
       @regvalue = {}
       hive.open(subkey, Win32::Registry::KEY_READ | access) do |reg|
-        from_string_to_wide_string(valuename) do |valuename_ptr|
-          if RegQueryValueExW(reg.hkey, valuename_ptr,
+        FFI::MemoryPointer.from_string_to_wide_string(valuename) do |valuename_ptr|
```

Comment:
Use `FFI::Pointer` here instead of `FFI::MemoryPointer`.  The helper is defined on the base class.

Reply:
Done